### ISSUE TITLE
Crash of demo_backpack_3d.launch

### ISSUE THREAD

**Issue by Murcamil** [googlecartographer/cartographer_ros#1356]
Hi!

I was running the 'demo_backpack_3d.launch' with RVIZ disabled(commented) with the Deutsches Museum 3D bag and got an error that I did not manage to solve. 
Because I read #87 , here is what I use: Intel NUC8i7BEH, 16GB RAM, with a clean install of Ubuntu 16.04.6 LTS, ros, cartographer. I observed no limitations reached in the System Monitor. 

While running `roslaunch cartographer_ros demo_backpack_3d.launch bag_filename:=${HOME}/Downloads/b3-2016-04-05-14-14-00.bag` 
I got the following error:
![Screenshot from 2019-07-12 14-35-57](https://user-images.githubusercontent.com/9487661/61128523-5f35b500-a4b2-11e9-8394-4d6163b9caac.png)

Because it took a long time to get to the error, I also tried to replay the bag faster with this edit in the _demo_backpack_3d.launch' file :
`  <node name="playbag" pkg="rosbag" type="play"
      args="--clock -r 10 $(arg bag_fil'name)" />` 
which would yield the error in the first 2 min. 

Thank you for the help

**Comment by Murcamil** [comment]
Running the same command on the same bag results in the same error, around the same bag time. BUT, I also noticed that I got, a bit before the error from the first post, the following error, which seems to come from the [ceres-solver](https://github.com/ceres-solver/ceres-solver/blob/master/internal/ceres/suitesparse.cc). 
![Screenshot from 2019-07-12 17-14-47](https://user-images.githubusercontent.com/9487661/61139535-cad74c80-a4c9-11e9-94e6-efa8f26bb1d7.png)

I can also add that loop closure does not work like #1357.
While Making the package, I saw that cartographer uses ceres-solver version 13.0 and that some files are not found during `src/cartographer/scripts/install_proto3.sh`
`-- Looking for pthread.h`
`-- Looking for pthread.h - found`
`-- Looking for pthread_create`
`-- Looking for pthread_create - not found`
`-- Looking for pthread_create in pthreads`
`-- Looking for pthread_create in pthreads - not found`

I have also tried reinstalling ceres-solver (succeeded in installing version 14) without solving my problems/ [https://github.com/googlecartographer/cartographer/issues/993#issuecomment-373482359](url)
